data/5e/wizards_of_the_coast/core/players_handbook/ph_classes.lst

```diff
@@ -20,54 +20,54 @@ CLASS:Barbarian	PREMULT:1,[!PRETEMPLATE:1,First Level],[PREABILITY:1,CATEGORY=In
 20	ABILITY:Barbarian Class Feature|AUTOMATIC|Barbarian ~ Primal Champion
 
 
-CLASS:Bard	SOURCEPAGE:p.51	HD:8		TYPE:PC.BASE	MAXLEVEL:20	DEFINE:BardLVL|0	BONUS:VAR|BardLVL|CL	DEFINE:POOL_Bard_Skill|0	DEFINE:BardToolCount|0	BONUS:VAR|POOL_Bard_Skill|1	BONUS:VAR|BardToolCount|1	ABILITY:Internal|AUTOMATIC|TYPE=ArmorProfLight	BONUS:ABILITYPOOL|Skill Proficiency|BardSkillCount	BONUS:ABILITYPOOL|Bard Tool Proficiency|BardToolCount
+CLASS:Bard	SOURCEPAGE:p.51	HD:8		TYPE:PC.BASE	MAXLEVEL:20	DEFINE:BardLVL|0	BONUS:VAR|BardLVL|CL	DEFINE:POOL_Bard_Skill|0	DEFINE:POOL_Bard_Magical_Secrets|0	DEFINE:BardToolCount|0	BONUS:VAR|POOL_Bard_Skill|1	BONUS:VAR|BardToolCount|1	ABILITY:Internal|AUTOMATIC|TYPE=ArmorProfLight	BONUS:ABILITYPOOL|Skill Proficiency|BardSkillCount	BONUS:ABILITYPOOL|Bard Tool Proficiency|BardToolCount
 CLASS:Bard	BONUS:VAR|SpellCasterTotalLVL|BardLVL	BONUS:VAR|SpellCasterClassCount|1
 CLASS:Bard	PREMULT:1,[!PRETEMPLATE:1,First Level],[PREABILITY:1,CATEGORY=Internal,Multiclass Allowed ~ Bard],[PRECLASS:1,Bard=1]
 CLASS:Bard	SPELLTYPE:Arcane	SPELLSTAT:CHA	BONUS:CASTERLEVEL|Bard|CL	MEMORIZE:NO
 ###Block
 1:REPEATLEVEL:1	DONOTADD:SKILLPOINTS
 ###Block
-1	CAST:1,1
-2	CAST:1,1
-3	CAST:1,1
-4	CAST:1,1
-5	CAST:1,1
-6	CAST:1,1
-7	CAST:1,1
-8	CAST:1,1
-9	CAST:1,1
-10	CAST:1,1
-11	CAST:1,1
-12	CAST:1,1
-13	CAST:1,1
-14	CAST:1,1
-15	CAST:1,1
-16	CAST:1,1
-17	CAST:1,1
-18	CAST:1,1
-19	CAST:1,1
-20	CAST:1,1
+1	CAST:0,2
+2	CAST:0,3
+3	CAST:0,4,2
+4	CAST:0,4,3
+5	CAST:0,4,3,2
+6	CAST:0,4,3,3
+7	CAST:0,4,3,3,1
+8	CAST:0,4,3,3,2
+9	CAST:0,4,3,3,3,1
+10	CAST:0,4,3,3,3,2
+11	CAST:0,4,3,3,3,2,1
+12	CAST:0,4,3,3,3,2,1
+13	CAST:0,4,3,3,3,2,1,1
+14	CAST:0,4,3,3,3,2,1,1
+15	CAST:0,4,3,3,3,2,1,1,1
+16	CAST:0,4,3,3,3,2,1,1,1
+17	CAST:0,4,3,3,3,2,1,1,1,1
+18	CAST:0,4,3,3,3,3,1,1,1,1
+19	CAST:0,4,3,3,3,3,2,1,1,1
+20	CAST:0,4,3,3,3,3,2,2,1,1
 ###Block
 1	KNOWN:0,4
 2	KNOWN:0,5
-3	KNOWN:0,5
-4	KNOWN:0,5
-5	KNOWN:0,5
-6	KNOWN:0,5
-7	KNOWN:0,5
-8	KNOWN:0,5
-9	KNOWN:0,5
-10	KNOWN:0,5
-11	KNOWN:0,5
-12	KNOWN:0,5
-13	KNOWN:0,5
-14	KNOWN:0,5
-15	KNOWN:0,5
-16	KNOWN:0,5
-17	KNOWN:0,5
-18	KNOWN:0,5
-19	KNOWN:0,5
-20	KNOWN:0,5
+3	KNOWN:0,4,2
+4	KNOWN:0,3,4
+5	KNOWN:0,2,4,2
+6	KNOWN:0,1,4,4
+7	KNOWN:0,0,4,4,2
+8	KNOWN:0,0,3,4,4
+9	KNOWN:0,0,2,4,4,2
+10	KNOWN:0,0,1,4,4,3
+11	KNOWN:0,0,0,4,4,3,2
+12	KNOWN:0,0,0,3,4,3,3
+13	KNOWN:0,0,0,2,4,3,3,2
+14	KNOWN:0,0,0,1,4,3,3,3
+15	KNOWN:0,0,0,0,4,3,3,3,2
+16	KNOWN:0,0,0,0,3,3,3,3,3
+17	KNOWN:0,0,0,0,2,3,3,3,3,2
+18	KNOWN:0,0,0,0,1,3,3,3,3,3
+19	KNOWN:0,0,0,0,0,3,3,3,3,4
+20	KNOWN:0,0,0,0,0,2,3,3,3,5
```

Comment:
I tried writing everything down in my commit notes. For this:
Bard Spellcasting: I have changed the way spell replacement works by simply using the Known: tag in the classes.lst file. I calculate the number of known spells operating under the assumption that for that level; one of the lowest spell slot levels will be replaced by the highest, as well as adding new known spells to the highest. For the bard, these numbers are adjusted to account for the fact that the magical secrets ability counts for the new spells learned during level progress (but not when it's part of the college progress line). The effect should be that with the fact that PCGen will allow a lower level spell to take the place of an available higher level one, any spell switching should work as the rules describe.